web-platform/website/src/services/AttributionService.tsx

```diff
@@ -27,27 +38,55 @@ export function AttributionProvider(props: { children: React.ReactNode }) {
   const [allAttributionCodes, setAllAttributionCodes] =
     useState<URLSearchParams>(new URLSearchParams());
 
-  async function loadAllAttributionCodes() {
-    // Add UTM codes to store links, if present.
-    let utmParams = new URLSearchParams();
-    if (typeof window !== "undefined") {
-      const utmKeys = ["source", "medium", "campaign", "term", "content"];
-
-      // Use "last-click" attribution, overwriting any stored UTM codes if new ones are encountered.
-      const storedParams = window.localStorage.getItem("rally_utm");
-      if (storedParams) {
-        utmParams = new URLSearchParams(storedParams);
+  const { userDocument, updateUserDocument, isDocumentLoaded } =
+    useUserDocument();
+
+  function loadAllAttributionCodes() {
+    function loadFromLocalStorage() {
+      // Add UTM codes to store links, if present.
+      let utmParams = new URLSearchParams();
+
+      if (typeof window !== "undefined") {
+        // Use "first-click" attribution, keeping any stored UTM codes if new ones are encountered.
+        const searchParams = new URL(window.location.href).searchParams;
+        utmKeys.forEach((key) => {
+          searchParams.has(key) &&
+            utmParams.set(key, searchParams.get(key) ?? "");
+        });
+
+        const storedParams = new URLSearchParams(
+          window.localStorage.getItem("rally_utm") || ""
+        );
+        if (storedParams) {
+          utmKeys.forEach((key) => {
+            storedParams.has(key) &&
+              utmParams.set(key, storedParams.get(key) ?? "");
+          });
+        }
+        window.localStorage.setItem("rally_utm", utmParams.toString());
       }
 
-      const searchParams = new URL(window.location.href).searchParams;
-      utmKeys.forEach((key) => {
-        const utmKey = `utm_${key}`;
-        searchParams.has(utmKey) &&
-          utmParams.set(utmKey, searchParams.get(utmKey) ?? "");
-      });
-      window.localStorage.setItem("rally_utm", utmParams.toString());
+      return utmParams;
     }
-    setAllAttributionCodes(utmParams);
+
+    let attribution = new URLSearchParams();
+    if (userDocument && userDocument.attribution) {
+      for (const utmKey in userDocument.attribution) {
+        attribution.set(utmKey, userDocument.attribution[utmKey]);
+      }
+    } else {
+      attribution = loadFromLocalStorage();
+      const attributionDoc: any = {};
+      attribution.forEach((value, key) => {
+        attributionDoc[key] = value;
+      })
+
+      updateUserDocument({
+        attribution: attributionDoc,
+      } as Partial<UserDocument>);
```

Comment:
We can check for user here and only do update doc if user is present.

Reply:
We can also refactor this as updateUserAttribution function and check to see if user exists and bail out early if user is not logged in.